FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Extensions
+{
+    public static class DictionaryExtensions
+    {
+        /// <summary>
+        /// Concat two string to int dictionary, for duplicated key, add the their value together.
+        /// For example. the concat result of dic1 {{"key1",1}} and dic2 {{"key1",2}, {"key2", 5}} is {{"key1",3}, {"key2", 5}}
+        /// </summary>
+        /// <param name="dic1">the first dictionary.</param>
+        /// <param name="dic2">the second dictionary.</param>
+        /// <returns>the concat dictionary.</returns>
+        public static Dictionary<string, int> ConcatDictionaryCount(this Dictionary<string ,int> dic1, Dictionary<string, int> dic2)
```

Comment:
An interesting scenario here: What if the `dic1` and `dic2` are equal? Say a naughty invoker calls this method like `myDict.ConcatDictionary(myDict)`, will this behaviour trigger infinite loop?